extensions/spark/kyuubi-spark-connector-hive/src/main/scala/org/apache/kyuubi/spark/connector/hive/write/HiveWriteBuilder.scala

```diff
@@ -47,45 +41,22 @@ case class HiveWriteBuilder(
       info,
       hiveTableCatalog,
       forceOverwrite,
-      mergePartitionSpec(),
-      ifPartitionNotExists)
+      dynamicPartitionSpec())
   }
 
   override def overwrite(filters: Array[Filter]): WriteBuilder = {
-    filters match {
-      case Array(AlwaysTrue) => // no partition, do nothing
-      case _ => staticPartition = deduplicateFilters(filters)
-    }
-    overwriteDynamicPartitions()
+    forceOverwrite = true
```

Comment:
why `filters` can be ignored here?

Reply:
The current DSv2 SupportsOverwrite#overwrite is designed to filter out (delete) data that matches deleteExpr when writer commit, But in the current catalyst implementation, these filterExprs are static partition EqualNullSafe expressions, KSHC does not currently require these processing filterExprs.

Reply:
can we add an `requires` guard here?